#!/bin/bash

#SBATCH --job-name rvar_call_rats
#SBATCH -A inbreh
#SBATCH -t 0-72:00:00
#SBATCH --nodes=1
#SBATCH --cpus-per-task=12
#SBATCH --mem=30G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=sharrin2@uwyo.edu
#SBATCH -e errs_outs/err_varcall_rats_%A_%a.err
#SBATCH -o errs_outs/std_varcall_rats_%A_%a.out
#SBATCH --array=1-199



# Variant calling all samples, split up by each scaffold


#load modules
module load gcc/12.2.0 bcftools/1.16

## Create directory path 
sorted=/project/getpop/ratsnakes/rat_bwa_rmd_out

## Set output directory & file prefix
OUT=/project/getpop/ratsnakes/rat_vcf/ratsnakes

#reference
REF=/project/getpop/ref/ratsnake_genome_files/ratsnake_scaffold/P_alleghaniensis_concatenate_ragtag.scaffolds.fasta

# Get the region of interest from the bed file
scaf=`cat /project/getpop/ref/ratsnake_genome_files/ratsnake_scaffold/P_alleghaniensis_concatenate_ragtag.scaffolds.bed | head -n $SLURM_ARRAY_TASK_ID | tail -n 1 | cut -f 1`

echo "started slurm job ${SLURM_ARRAY_TASK_ID}: scaffold $scaf"



# Go to target directory
cd $sorted

#run mpileup
bcftools mpileup --threads 12 -A -a DP,AD --regions $scaf -f $REF *.rmd.bam | bcftools call -f GQ,GP --multiallelic-caller --variants-only --threads 12 -O z -o ${OUT}_$scaf.vcf.gz

echo "completed slurm job ${SLURM_ARRAY_TASK_ID}: scaffold $scaf"